fdbclient/TenantManagement.actor.cpp

```diff
@@ -37,4 +38,30 @@ TenantMode tenantModeForClusterType(ClusterType clusterType, TenantMode tenantMo
 	}
 }
 
+int64_t extractTenantIdFromMutation(MutationRef m) {
+	ASSERT(!isSystemKey(m.param1));
+
+	if (isSingleKeyMutation((MutationRef::Type)m.type)) {
+		// The first 8 bytes of the key of this OP is also an 8-byte number
+		if (m.type == MutationRef::SetVersionstampedKey && m.param1.size() >= 4 &&
+		    parseVersionstampOffset(m.param1) < 8) {
+			return TenantInfo::INVALID_TENANT;
```

Comment:
Just curious, what is the expected behavior w.r.t encryption here? Does `invalid_tenant` gets mapped to `default_domain`? Reason I ask is, I was under assumption that for a `cluster with domainAware encryption`, blocks can be encrypted with either: `systemKey` and/or `per-tenant` EK.

Reply:
Also, it would be worth putting such an assert to catch fallouts (which maybe legit, hence, updating our understanding...)

Reply:
Confirmed with AJ that in required mode we cannot modify the first 8 bytes of a key (so encryption is fine as the tenant only matters in required tenant mode). This code is shared with the validation work that xiaoxi did so that's why we are doing this check here (she needs to validate that in required mode we aren't modifying the first 8 bytes, but our use case doesn't really need this)

Reply:
Great! 
I think it makes the point of having an assert in `encryption` modules more pertinent; for `tenant::required` `domainId` should be either `SYSTEM_KEYSPACE_ENCRYPT_DOMAIN_ID` or `> 0` (i.e. potentially valid tenant). 
We need not to do it as part of this PR, but, may be some followup PR(s)

Thoughts?

Reply:
I agree we should do this check (clear range for now will violate this). Maybe @sfc-gh-yiwu can add this as part of the pr when we modify how we fetch encryption keys in the CP